dev/tests/static/framework/Magento/CodeMessDetector/Rule/UnusedCode/UnusedFormalParameter.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CodeMessDetector\Rule\UnusedCode;
+
+use PHPMD\AbstractNode;
+use PHPMD\Node\ClassNode;
+use PHPMD\Node\MethodNode;
+use PHPMD\Rule\UnusedFormalParameter as PhpmdUnusedFormalParameter;
+
+class UnusedFormalParameter extends PhpmdUnusedFormalParameter
+{
+    /**
+     * This method collects all local variables in the body of the currently
+     * analyzed method or function and removes those parameters that are
+     * referenced by one of the collected variables.
+     *
+     * @param AbstractNode $node
+     * @return void
+     */
+    protected function removeUsedParameters(AbstractNode $node)
+    {
+        parent::removeUsedParameters($node);
+        $this->removeVariablesUsedInPlugins($node);
+    }
+
+    /**
+     * Remove required method variables used in plugins from given node
+     *
+     * @param AbstractNode $node
+     */
+    private function removeVariablesUsedInPlugins(AbstractNode $node)
+    {
+        if (!$node instanceof MethodNode) {
+            return;
+        }
+
+        /** @var ClassNode $classNode */
+        $classNode = $node->getParentType();
+        if (!$this->isPluginClass($classNode->getNamespaceName())) {
+            return;
+        }
+
+        /**
+         * Around and After plugins has 2 required params $subject and $proceed or $result
+         * that should be ignored
+         */
+        foreach (['around', 'after'] as $pluginMethodPrefix) {
+            if ($this->isFunctionNameStartingWith($node, $pluginMethodPrefix)) {
+                $this->removeVariablesByCount(2);
+
+                break;
+            }
+        }
+
+        /**
+         * Before plugins has 1 required params $subject
+         * that should be ignored
+         */
+        if ($this->isFunctionNameStartingWith($node, 'before')) {
+            $this->removeVariablesByCount(1);
+        }
+    }
+
+    /**
+     * Check if the first part of function fully qualified name is equal to $name
+     *
+     * @param MethodNode $node
+     * @param string $name
+     * @return boolean
+     */
+    private function isFunctionNameStartingWith(MethodNode $node, string $name): bool
+    {
+        return (0 === strpos($node->getImage(), $name));
```

Comment:
Interesting... why it's named as getImage? 😃

Reply:
Both `getName` and `getImage` contain `return $this->name`.
Tests in phpmd source use `getImage`

Reply:
Could you add this as a comment?